plugins/javadoc/src/main/kotlin/org/jetbrains/dokka/javadoc/signatures/JavadocSignatureProvider.kt

```diff
@@ -150,7 +150,7 @@ class JavadocSignatureProvider(ctcc: CommentsToContentConverter, logger: DokkaLo
                 annotationsBlock(t)
             }
             signatureWithoutModifiers {
-                text(t.name)
+                text(t.name) // Investigate if java classes can be somehow variant
```

Comment:
Then it should be something like that `? extends T extends SomeInterface`

Reply:
On the other hand we should somehow indicate that when the class comes from Kotlin sources

Reply:
I think `? extends SomeInterface` or `? super SomeInterface`  or am I wrong?  🤔

Reply:
Yeah, but that works for properties/parameters. There is no such mechanism for class declaration, which we here discuss.

Reply:
I mean, in Kotlin we can have `class Klass<out T : SomeInterface>`. To accomplish that in Java, you should have something like `class Klass<? extends T extends SomeInterface>()` which is of course not a correct syntax.